### PR TITLE
Move builtin loading code entirely out of __init__

### PR DESCRIPTION
**Pull request by rocky**
... YAY!

This paves the way for further improvements.

### CODE REVIEW

File: mathics/version.py [hunk 1]
```diff
@@ -5,4 +5,4 @@
 # well as importing into Python. That's why there is no
 # space around "=" below.
 # fmt: off
-__version__="6.0.2dev0"  # noqa
+__version__="6.1.0dev0"  # noqa
```
Comment (rocky) [inline mathics/version.py:8]: This probably should be 7.0.0. Even before this change there were a number of API breaking changes. You'll see them in the recent Mathics3 Modules (natlang, and graph). And we'll have to change these more for this as well.

File: mathics/core/load_builtin.py [hunk 2]
```diff
@@ -16,6 +17,11 @@
 from mathics.core.pattern import pattern_objects
 from mathics.core.symbols import Symbol
 from mathics.eval.makeboxes import builtins_precedence
+from mathics.settings import ENABLE_FILES_MODULE
+
+# List of Mathics3 Builtin modules.
+# This is initialized via below import_builtins modules
+modules = []
```
Comment (rocky) [inline mathics/core/load_builtin.py:24]: modules is a too vague a name. Probably mathics3_builtin_modules is better.

File: mathics/main.py [hunk 3]
```diff
@@ -24,13 +24,16 @@
...
 from mathics.core.expression import Expression
+from mathics.core.load_builtin import import_and_load_builtins
 from mathics.core.parser import MathicsFileLineFeeder, MathicsLineFeeder
 from mathics.core.read import channel_to_stream
 from mathics.core.rules import BuiltinRule
 from mathics.core.streams import stream_manager
 from mathics.core.symbols import SymbolNull, strip_context
 from mathics.timing import show_lru_cache_statistics
 
+import_and_load_builtins()
```
Comment (rocky) [inline mathics/main.py:35]: Now that this section is isolated we could and should be able to time how long running this subroutine takes.

File: mathics/core/load_builtin.py (hunk 2 above)
Comment (mmatera) [reply mathics/core/load_builtin.py:24]: agree

**Review by mmatera — APPROVED** [review]
All the changes look good. Merge this when you feel it is ready.

**Comment by rocky** [comment]
running `mathics -e '1+1' ` with the timing decorator around loading builtin modules show that 2.2 seconds out of 7 elapsed are spent in loading. 

Next changes will be to be able to load builtin modules one at a time rather than as a set of sequential which do phases of module loading.

**Pull request merged**